types/rest/rest.go

```diff
@@ -232,6 +232,8 @@ func ParseQueryHeightOrReturnBadRequest(w http.ResponseWriter, cliCtx context.CL
 		if height > 0 {
 			cliCtx = cliCtx.WithHeight(height)
 		}
+	} else {
+		cliCtx = cliCtx.WithHeight(0)
```

Comment:
/cc @colin-axner I had to do this because somehow the height from _previous_ queries will still attached to the context. I have zero clue how this happens (maybe its only through tests?). Regardless, this fixes it.

Reply:
ohhh, I think its because the [`RestServer`](https://github.com/cosmos/cosmos-sdk/blob/f218de8141826400e79963fc10446b7df7be39cb/client/lcd/root.go#L30) has the cliCtx field that I think never changes so any updates to it persist as long as you don't restart your server. We might want to reset the height somewhere else, or ensure `ParseQueryHeightOrReturnBadRequest` is always called on every request

Reply:
> Edit: This still doesn't make sense to me because the rest server would need the cliCtx to be returned back

Exactly...I'm dumbfounded here. Maybe it has something to do with the way closures work in go. But that doesn't make any sense either because this only happened on _certain_ endpoints.